cryptographic_estimators/MREstimator/mr_algorithm.py

```diff
@@ -0,0 +1,101 @@
+# ****************************************************************************
+# Copyright 2023 Technology Innovation Institute
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+# ****************************************************************************
+
+
+from ..base_algorithm import BaseAlgorithm
+from .mr_problem import MRProblem
+
+
+class MRAlgorithm(BaseAlgorithm):
+    def __init__(self, problem: MRProblem, **kwargs):
+        """
+        Base class for MR algorithms complexity estimator
+
+        INPUT:
+
+        - ``problem`` -- MRProblem object including all necessary parameters
+
+        """
+        super(MRAlgorithm, self).__init__(problem, **kwargs)
+        w = kwargs.get("w", 2.81)
+        theta = kwargs.get("theta", 2)
+        self._name = "BaseMRAlgorithm"
+
+        if  w < 2 or 3 < w:
+            raise ValueError("w must be in the range 2 <= w <= 3")
+
+        if theta is not None and (theta > 2 or theta < 0):
+                raise ValueError("theta must be None or an integer in the range 0 <= theta <= 2")
```

Comment:
How can theta be None, in line 35 it is set to the default value (2) or are you expecting the user to actively pass theta=None?

Reply:
yes, `tetha = None` means to capture the specific case (commonly) used by the community where the bit-conversion factor is set to be `2*log2(q) **2 + log2(q)`

Reply:
ok, hope that is documented somewhere? 😅